### PR TITLE
Add OnPrepareResponse option w/resp header to static files doc

### LINKED ISSUE #2625: Document OnPrepareResponse in the Static Files doc
[Working with static files in ASP.NET Core (fundamentals/static-files)](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/static-files) doesn't document `OnPrepareResponse`; consequently, devs can't see how to add a header. For example, https://github.com/aspnet/ResponseCaching/issues/97.

Might be able to roll it into the last code example of *Serving static files*. Could go something like the following with some text added to call it out ...
```c#
public void Configure(IApplicationBuilder app)
{
    app.UseStaticFiles();

    app.UseStaticFiles(new StaticFileOptions()
    {
        FileProvider = new PhysicalFileProvider(
            Path.Combine(Directory.GetCurrentDirectory(), @"MyStaticFiles")),
        RequestPath = new PathString("/StaticFiles"),
        OnPrepareResponse = ctx =>
        {
            ctx.Context.Response.Headers.Append("Cache-Control", "max-age=3600");
        }
    });
}
```
Let me know if you want me to take a stab at it.

[Side Note 1: I think I found a bug tho in the Static File Middleware. I'll open an issue to ask.]

[Side Note 2: `IHostingEnvironment` and `ILoggerFactory` are injected into `Configure()` methods but not used. Seems like they should be removed.]

### PR DESCRIPTION
**Pull request by GuardRex**
Fixes #2625

### CODE REVIEW

File: aspnetcore/fundamentals/static-files.md [hunk 1]
```diff
@@ -59,6 +59,12 @@ For a request to access *test.png*, configure the static files middleware as fol
 
 A request to `http://<app>/StaticFiles/test.png` will serve the *test.png* file.
 
+Using `StaticFileOptions()`, you can also set response headers. For example, you can add a `Cache-Control` header, as you see here making the static files served from the *wwwroot* folder publicly cacheable for 10 minutes (600 seconds):
```
Comment (Rick-Anderson) [inline aspnetcore/fundamentals/static-files.md]: consider:
`StaticFileOptions()` can set response headers. For example, the code below allows static files to be served from the *wwwroot* folder publicly cacheable for 10 minutes (600 seconds):

or maybe 
`StaticFileOptions()` can also set response headers.

Comment (GuardRex) [inline aspnetcore/fundamentals/static-files.md]: Try this one. If it still doesn't work, I'll drop back to just ...

>StaticFileOptions() can also set response headers:

**Comment by Rick-Anderson** [comment]
Really nice work!

**Comment by tinchou** [comment]
Appreciate it!

**Pull request merged**